src/main/java/com/ait/lienzo/client/core/util/Geometry.java

```diff
@@ -1667,7 +1667,7 @@ public static Point2D findCenter(BoundingBox box)
         return new Point2D(box.getX() + box.getWidth() / 2, box.getY() + box.getHeight() / 2);
     }
 
-    private static Point2D getProjection(Point2D center, Point2D intersection, double length)
+    public static Point2D getProjection(Point2D center, Point2D intersection, double length)
     {

```

Comment:
Since the method is public now, you can insert validations to the attributes: center, intersection, length.